app/forms/portal/sign_form8879.rb

```diff
@@ -0,0 +1,59 @@
+module Portal
+  class SignForm8879 < Form
+    attr_accessor :primary_accepts_terms, :primary_confirms_identity, :ip, :transaction_failed
+
+    def initialize(tax_return, params = {})
+      @tax_return = tax_return
+      super(params)
+    end
+
+    def sign
+      errors.add(:primary_accepts_terms, :blank) && return unless primary_accepts_terms == "yes"
+      errors.add(:primary_confirms_identity, :blank) && return unless primary_confirms_identity == "yes"
+      return true if ActiveRecord::Base.transaction do
+        record_primary_signer_information_on_tax_return
+        change_tax_return_status
+        @tax_return.save!
+        create_signed_8879
+        true
+      end
+
+      errors.add(:transaction_failed)
```

Comment:
how do we get away with not returning a falsey value from this method and having it work?